src/groth16/aggregate/verify.rs

```diff
@@ -258,6 +258,11 @@ fn verify_tipp_mipp<E: Engine, R: rand::RngCore + Send>(
         // U = e(A,v2)
         let _check_u = pairing_checks.merge_miller_inputs(&[(final_c,&fvkey.1)],final_uc)
     };
+    if let Err(_) = final_z {
+        pairing_checks.invalidate();
```

Comment:
This is to avoid the situation where I need to make `verify_tipp_mipp` return a result and then change all the channels to handle results instead. This is slightly misleading as this "error" as this error is raised when the vectors are not of the right size, and not when the equations don't match, but I'm quite reluctant as to change again most of the code related to pairing checks just for this.

Reply:
If you do this, then you have also reversed the previous 'pure' stance that this short-circuiting 'false' in `PairingCheck` verification is strictly about the boolean outcome of well-formed checks. This is already hacked into `PairingCheck`s (since it's _not_ really a pairing check) as an expedient. I don't think it should be abused further. An error associated with malformed input really should be propagated to the caller.